Periodic_2_triangulation_2/include/CGAL/draw_periodic_2_triangulation_2.h

```diff
@@ -23,255 +26,196 @@
 
 namespace CGAL {
 
-// Default color functor; user can change it to have its own face color
-struct DefaultColorFunctorP2T2 {
-  template <typename P2T2>
-  static CGAL::IO::Color run(const P2T2 &,
-                         const typename P2T2::Periodic_triangle_iterator /*ti*/) {
-    //CGAL::Random random((unsigned int)(std::size_t)(&*ti));
-    //return get_random_color(random);
-    return CGAL::IO::Color(73, 250, 117);
-  }
+namespace draw_function_for_P2T2
```

Comment:
Maybe in this case we need to keep the class SimplePeriodic2Triangulation2ViewerQt ? Otherwise hhow can you keep the old behavior ?

Reply:
Do you mean regarding continuously changing color? if so, I think we can pass a nonconst drawing_functor.

What do you think?

Reply:
I mean regarding the different key shortcuts used to display some specific elements. In your modifications, these shortcuts are for now removed. You can either re-introduce the class SimplePeriodic2Triangulation2ViewerQt to re-add these shortcurts, or maybe it is possible to add an external slot but I am not sure this is possible.

Reply:
**WIP**, using std::function.